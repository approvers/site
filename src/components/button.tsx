import { MouseEvent, ReactNode, Ref, forwardRef } from "react";
import styles from "../scss/components/button.module.scss";

export interface ButtonProps {
  children: ReactNode;
  onClick?: (event: MouseEvent<HTMLDivElement>) => void;
}

const button = ({ onClick, children }: ButtonProps, ref?: Ref<HTMLDivElement>): JSX.Element => (
  <div ref={ref} onClick={onClick} className={styles.button}>
    {children}
  </div>
);

export const Button = forwardRef(button);
