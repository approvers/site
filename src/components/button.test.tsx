import { Button } from "./button";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Button
        onClick={() => {
          alert("Clicked!");
        }}
      >
        Click Here
      </Button>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
