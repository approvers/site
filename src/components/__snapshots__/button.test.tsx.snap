// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="button"
  onClick={[Function]}
>
  Click Here
</div>
`;
